Standardise `current_user` call within Commands tasks (Meetings)

greenwoodt opened the pull request:
#### :tophat: What? Why?
```current_user``` is standardized within specific tasks of various commands in the Meetings module. 

#### :pushpin: Related Issues
- Related to #10199 

#### Testing
Run rspec ```decidim-meetings/spec/commands/create_answer_spec.rb```  + ```decidim-meetings/spec/commands/update_meeting_spec.rb```

### :camera: Screenshots

:hearts: Thank you!

andreslucena commented on decidim-meetings/app/commands/decidim/meetings/admin/update_question_status.rb:
You need to delete the attr_reader for :current_user in line 34 in this file

andreslucena commented on decidim-meetings/spec/commands/admin/update_question_status_spec.rb:
I think you can remove `let(:current_user) { create(:user, organization: meeting_component.organization) }` in line 10 (and also current_organization in line 9, as it doesn't seem to be used)

greenwoodt replied:
Will give it a go!

andreslucena reviewed (changes requested):
I have some suggestions here, can you check them out please?

greenwoodt commented:
@andreslucena does the :current_user actually get delegated to the :form in ```decidim-meetings/app/commands/decidim/meetings/admin/update_question_status.rb```? 

These errors indicate to me that it doesn't recognize ```:form```?

andreslucena commented:
> @andreslucena does the :current_user actually get delegated to the :form in `decidim-meetings/app/commands/decidim/meetings/admin/update_question_status.rb`?
> 
> These errors indicate to me that it doesn't recognize `:form`?

Delegation works by saying "when you have to call a_method ask for this to `an_object` directly, instead of me having to ask for `an_object.a_method`".

So, in our case what we're saying is "when you have to call `current_user` ask for this to `form` directly, instead of me having to ask for `form.current_user`".

This is always starting from the object `form`, so that need to be in the initializer method (i.e. passed as an argument to `new`). If you don't have the object `form` as an argument it can't be delegated, so that particular change needs to be undone.

greenwoodt commented:
> > @andreslucena does the :current_user actually get delegated to the :form in `decidim-meetings/app/commands/decidim/meetings/admin/update_question_status.rb`?
> > These errors indicate to me that it doesn't recognize `:form`?
> 
> Delegation works by saying "when you have to call a_method ask for this to `an_object` directly, instead of me having to ask for `an_object.a_method`".
> 
> So, in our case what we're saying is "when you have to call `current_user` ask for this to `form` directly, instead of me having to ask for `form.current_user`".
> 
> This is always starting from the object `form`, so that need to be in the initializer method (i.e. passed as an argument to `new`). If you don't have the object `form` as an argument it can't be delegated, so that particular change needs to be undone.

Then its important we remove the comment line 13 from ```decidim-meetings/app/commands/decidim/meetings/admin/update_question_status.rb``` in the future as it easily misleads @andreslucena.

andreslucena commented on decidim-meetings/app/commands/decidim/meetings/admin/update_question_status.rb:
We need to *not* change this class (and the controller that's calling it, and the spec).

Notice that we aren't passing the `form` argument in the initializer, so it can't be delegated to it. Please leave it out as it was before.

andreslucena commented:
> Then its important we remove the comment line 13 from `decidim-meetings/app/commands/decidim/meetings/admin/update_question_status.rb` in the future as it easily misleads

Yes, I agree, that line is a bug for sure!

greenwoodt commented:
I have reverted previous changes and removed the comment line causing confusion in ```decidim-meetings/app/commands/decidim/meetings/admin/update_question_status.rb```. 

We still have the same error as before on line 79 in ```spec/system/live_meeting_answer_questions_spec.rb:79``` with regards to a replied question in live meeting answers.

andreslucena commented on decidim-meetings/app/controllers/decidim/meetings/polls/answers_controller.rb:
```diff
@@ -13,7 +13,7 @@ def create
           enforce_permission_to(:create, :answer, question:)
           @form = form(AnswerForm).from_params(params, question:, current_user:)
```
This needs to be changed so the `current_user` is available in the form instance.

```suggestion
          @form = form(AnswerForm).from_params(params.merge(question:, current_user:))
```

andreslucena commented on decidim-meetings/app/commands/decidim/meetings/admin/update_question_status.rb:
```diff
@@ -10,7 +10,6 @@ class InvalidStatus < StandardError; end
         # Initializes a UpdateQuestionnaire Command.
         #
-        # form - The form from which to get the data.
         # questionnaire - The current instance of the questionnaire to be updated.
```
```suggestion
        # questionnaire - The current instance of the questionnaire to be updated.
        # current_user  - The current user.
```

andreslucena reviewed (changes requested):
Only one missing comment and we're ready to merge
